src/nullablevector.jl

```diff
@@ -311,3 +313,27 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+function vcat(A::AbstractVector, B::AbstractVector...)
```

Comment:
I think defining these functions would be a regression compared to just leaving the current functionality as is.
```julia
# this rule seems more intuitive
isa(array[1], NullableArray) ? NullableArray{T} : Array[Nullable{T}]

# than this rule
isa(array[1], NullableArray) ||
isa(array[2], NullableArray) ||
isa(array[3], NullableArray) ? NullableArray{T} : Array[Nullable{T}]
```

If the general functionality is type piracy then I feel the best way forward would be to either address https://github.com/JuliaLang/julia/issues/2326 directly or accept the current behavior. Thoughts?